.github/workflows/ci.yml

```diff
@@ -1,26 +0,0 @@
-name: Avocado Gating CI
```

Comment:
I think we shouldn't delete those checks. If I remember correctly, the idea is to run a style check before the unit and functional tests and if it fails, the other tests wouldn't be run. Unfortunately, I wasn't able to find an issue about it. So removing this file is going backwards in this effort. Maybe @beraldoleal  can give more info.

Reply:
Yes @richtja.

The original ideal in order to save some time running those checks that we discussed is having a nice pipeline (for instance: style checks -> unit -> functional -> smoke checks) with some dependencies.

If styles and signed-off tests are failing, then we could exit and save some time. It is a tradeoff but with most of the CI systems having high price/compute hour, we concluded that we would give this approach a try, and see what happens.

This stage must be there, and being the first of all others. if this fails, then we won't proceed to the next stages.

Reply:
Yes @richtja.

The original ideal in order to save some time running those checks that we discussed is having a nice pipeline (for instance: style checks -> unit -> functional -> smoke checks) with some dependencies.

If styles and signed-off tests are failing, then we could exit and save some time. It is a tradeoff but with most of the CI systems having high price/compute hour, we concluded that we would give this approach a try, and see what happens.

This stage must be there, and being the first of all others. if this fails, then we won't proceed to the next stages.